### PR TITLE
XRLightProbe: Use single entry for event

### PR DESCRIPTION
**Pull request by Elchi3**
#### Summary
Companion PR for https://github.com/mdn/content/pull/10373. See the discussion at https://github.com/mdn/content/discussions/9098

### CODE REVIEW

File: api/XRLightProbe.json [hunk 1]
```diff
@@ -48,55 +48,6 @@
           "deprecated": false
         }
       },
-      "onreflectionchange": {
-        "__compat": {
-          "spec_url": "https://immersive-web.github.io/lighting-estimation/#dom-xrlightprobe-onreflectionchange",
```
Comment (hamishwillee) [inline api/XRLightProbe.json:53]: @Elchi3 If we're going to have a single entry for both the event and its handler, should the spec here moved down to reflectionchange_event ? (i.e. it has an array of specs)?

Comment (Elchi3) [reply api/XRLightProbe.json:53]: Yes, I think this is a great idea!

**Comment by ddbeck** [comment]
@Elchi3 OK, so this suggests something very interesting, but this implies something that's slightly surprising to me: we will be omitting exposed interfaces that web developers can find and that tooling can inspect. Should we worry about that? Do we forbid any other things that are developer-exposed names? Are we OK with that?

I think if we're going to merge something like this, we ought to have data guideline that explains exactly which bits of IDL we're ruling out (e.g., "Don't create features for `attribute EventHandler on*`).

**Comment by Elchi3** [comment]
Yes, I believe this is what we would like to do. @foolip what do you think about this?

See also https://github.com/mdn/browser-compat-data/pull/12510 which I believe goes in the same direction and suggests the same consequences.

**Comment by foolip** [comment]
We need a guideline for this, yes. It wouldn't be the first bit of IDL that excluded though, there's already https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines.md#constants

**Comment by ddbeck** [comment]
I forgot about the constants case. Yeah, I'm much more comfortable with this, given the precedent. I would like to see an accompanying guideline before any of these merge, so that I have something to point to in the release notes.

**Comment by Elchi3** [comment]
Opened https://github.com/mdn/browser-compat-data/pull/13595 to discuss the data guideline change.

**Comment by Elchi3** [comment]
Content updates merged and new events guideline is in place. This is ready.

**Review by queengooborg — APPROVED** [review]

**Comment by foolip** [comment]
Post-merge review in light of https://github.com/mdn/browser-compat-data/pull/13924#issuecomment-1005498544.

Here the versions of the on* properties matched the version of the interface, so there's nothing to research. (It *is* possible that the event wasn't actually fired, but that's a less common situation than missing on* properties.)

**Pull request merged**